include/cell/compute/softmax.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "cell/compute/reduce.hpp"
+#include "cuda_utils.hpp"
+#include "types/layout.hpp"
+#include "types/mod.hpp"
+#include "types/tile_shape.hpp"
+
+namespace tiledcuda::cell::compute {
+
+namespace tl = tile_layout;
+
+template <typename RegTile, const tl::Layout kLayout>
+struct Softmax {
+    using DType = typename RegTile::DType::DType;
+    using BaseShape = traits::BaseTileShape<DType>;
+
+    static constexpr int kRows = RegTile::kRows;
+    static constexpr int kCols = RegTile::kCols;
+};
+
+template <typename RegTile>
+struct Softmax<RegTile, tl::Layout::kRowMajor> {
+    using DType = typename RegTile::DType::DType;
+    using BaseShape = traits::BaseTileShape<DType>;
+
+    static constexpr int kRows = RegTile::kRows;
+    static constexpr int kCols = RegTile::kCols;
+
+    template <typename ReduceTile>
+    DEVICE void operator()(RegTile& tile, ReduceTile& reduce_tile) {
+#pragma unroll
+        for (int i = 0; i < kRows; ++i) {
+#pragma unroll
+            for (int j = 0; j < kCols; ++j) {
+                auto base_tile = tile(i, j);
+
+                tile(i, j)(0, 0) = exp(base_tile(0, 0));
+                tile(i, j)(0, 1) = exp(base_tile(0, 1));
+                tile(i, j)(1, 0) = exp(base_tile(1, 0));
+                tile(i, j)(1, 1) = exp(base_tile(1, 1));
+                tile(i, j)(0, 2) = exp(base_tile(0, 2));
+                tile(i, j)(0, 3) = exp(base_tile(0, 3));
+                tile(i, j)(1, 2) = exp(base_tile(1, 2));
+                tile(i, j)(1, 3) = exp(base_tile(1, 3));
+            }
+        }
+
+        compute::SumReduce<RegTile, tl::Layout::kRowMajor> row_sum;
+        row_sum(tile, reduce_tile);
+
+        // __syncthreads();
```

Comment:
Is line 52 necessary? If it is no longer useful, it could be deleted.

Reply:
Deleted.